### DISCUSSION TITLE
docs/cli/gen-dao

### DISCUSSION THREAD

**Discussion by giscus** [gogf/gf-site#27] [General]
# docs/cli/gen-dao

gen dao 命令是 CLI 中最频繁使用、也是框架设计的工程规范能否准确落地的关键命令。该命令用于生成 dao 数据访问对象、 do 数据转化模型及 entity 实例数据模型 Go 代码文件。由于该命令的参数、选项较多，我们推荐使用配置文件来管理生成规则。

https://goframe.org/docs/cli/gen-dao

**Comment by remfath** [comment]
gf gen dao 生成的结构中 id、url、ip等，可以自定义为 ID、URL、IP 吗？ 这种更符合意思。

**Reply by remfath** [reply]
现在都是生成的 Id、Url、Ip，这看着很奇怪

**Reply by gqcn** [reply]
> 现在都是生成的 Id、Url、Ip，这看着很奇怪

可以试试`jsonCase`参数，设置为`SnakeFirstUpper`试试呢？

**Reply by remfath** [reply]
不是json tag呢，比如表 user 有 user_id，希望生成的结构体为：
```go
type User struct {
	UserID string
}
```
而不是
```go
type User struct {
	UserId string
}
```
目前没有看到有地方可以调整，我看源码不支持这个配置。

**Reply by gqcn** [reply]
> 不是json tag呢，比如表 user 有 user_id，希望生成的结构体为：
> 
> ```go
> type User struct {
> 	UserID string
> }
> ```
> 
> 而不是
> 
> ```go
> type User struct {
> 	UserId string
> }
> ```
> 
> 目前没有看到有地方可以调整，我看源码不支持这个配置。

嗯，这个不支持，比较偏定制化了。

**Comment by xiaoyi510** [comment]
怎么tinyint 1也整到 bool 了 如何关闭此特性?

**Reply by xiaoyi510** [reply]
框架过多的干涉了 tinyint了  一般作为状态值 或者某些 1-9类型的 都可能是使用tinyint 按照以前的bit(1) 就行了吧

**Reply by gqcn** [reply]
主要是mysql针对boolean类型就是用的tinyint(1)，这块我们再评估下。

**Reply by xiaoyi510** [reply]
或者考虑增加配置项来让开发者决策,昨日更新后,,又将数据库的表全部改了一次~~

类似  type[收款,付款] status [待使用,已过期,已使用] 这种使用tinyint 1 场景较多

**Reply by gqcn** [reply]
是的，还是`bit(1)`比较适合`bool`类型，`tinyint(1)`不太合适。`gen dao`不支持从`tinyint(1)`识别`bool`。

**Reply by buexplain** [reply]
如果数据类型无法约束用户的使用方式，那就不要猜测用户的使用方式。
怎么才能扭转成int？我现在重新生成后，项目都跑不起来了。

**Comment by 1990simon** [comment]
hello 请教下, 看了2.7 2.8 没看到 data model 自动创建 sql 表的描述, 只看到了 db table 转 dao model, 是不支持吧? 后续能否支持下? 比如比较流行的框架是支持的, 开发时直接定义 data model 多方便啊, 然后run起来就创建了表如果不存在, 这种是比较方便主流的开发流程了吧.
谢谢各位开发者.

**Reply by gqcn** [reply]
你好，暂时没有`migrate`功能，如果有好的想法也可以提`issue`讨论。

**Comment by jarry8359** [comment]
// 修复生成的entity对象json标签大小写问题
文件位置：gogf/internal/cmd/gendao/gendao_structure.go

func generateStructFieldDefinition(
    ctx context.Context, field *gdb.TableField, in generateStructDefinitionInput,
) (attrLines []string, appendImport string) {
  var (
    ...
    // 修改以下代码，可以修复jsonCase：CamelLower问题
    jsonTag = gstr.CaseConvert(gstr.ToLower(field.Name), gstr.CaseTypeMatch(in.JsonCase))
  )      
  ...
}

**Comment by jarry8359** [comment]
希望gf gen dao生成的entity对象可以添加dc数据库字段注释，要不然swagger里字段没有中文。比如：
Id string `json:"id" orm:"ID" dc:"主键"`

**Reply by xiaoyi510** [reply]
![image](https://github.com/user-attachments/assets/5e1079ad-c903-47cf-89bc-73aed6956302)
有的啊

**Comment by chenh034** [comment]
在fieldMapping的配置内容中， 除了type和import， 是否可以增加一个配置项用于控制该字段的jsonTag细节，比如是否添加omitempty

**Comment by wu-weichao** [comment]
旧版本生成
TASKNAME     string      `json:"tASKNAME"     orm:"TASK_NAME_"      `

新版本 2.9.0 生成
TaskName       string      `json:"tASKNAME"          orm:"TASK_NAME_"    ` 

有办法让2.9.0版本生成的属性名和旧版本一致吗？期望属性名称还是 TASKNAME

**Comment by i-bing** [comment]
tables & tablesEx 是否可以支持Pattern ？ 比如 user_* 当表太多，需要输入太多了。

**Reply by dmwin72015** [reply]
确实是！太多表写在一起，而且查看起来也不方便，也不知道有没有什么更方便的办法！
